### DISCUSSION TITLE
Adding classpath entries to built Quarkus application during startup

### DISCUSSION THREAD

**Discussion by GavinRay97** [quarkusio/quarkus#44579] [Q&A]
### Description

I have a Quarkus app which uses the Apache Phoenix JDBC driver.

This driver wants certain config resources on the application classpath.
Unfortunately, these config resources must be provided by end-users of the application, so they are not available at build-time.

Is there any way I can include more entries in the Quarkus application classpath when starting my application?

Thank you =)

**Comment by gsmet** [comment]
(Moved to a discussion)

That's definitely not the ideal situation for Quarkus. I think the only option will be [reaugmentation](https://quarkus.io/guides/reaugmentation).

You can't initialize provide the information to the driver programmatically?

**Reply by GavinRay97** [reply]
Yes, it's certainly not ideal

This usecase has actually popped up in 2 instances now:

1. The client wants to use our tool with a proprietary JDBC jar that is a dialect of an existing DB. 
2. A JDBC driver requires runtime configuration via classpath entries that are end-user supplied.

For the 1st usecase, we ended up giving them a custom `build.gradle.kts` that copied `./lib/vendored` jars during build time, and asked them to build the image themselves.

For the second usecase, due to technical limitations, we unfortunately cannot do this.

From what I've read, there doesn't seem to be much in the way of extending/modifying Quarkus app classpaths after build.

What about using an `uber-jar` and manually specifying `-cp` when launching?
Because then, all of the classpath entries would be in the `.jar` so it'd be alright to override the `-cp` flag.
